### PR TITLE
Process string escapes properly.

### PR DESCRIPTION
**Pull request by ltratt**
Some of these are hard to test in practise (backspace and form feed are not very portable these days). The spec can only be found in the SOM test suite.

### CODE REVIEW

**Comment by Hirevo** [comment]
An interesting thing about string escapes in SOM that I found surprising is that, in case of a unknown escaped character (like `\a`), implementations seems to silently discard the character:
```plain
---> 'abc\adef'
it = abcdef
---> 'abc\adef' length
it = 6
```
Both the C and Java implementations simply ignore that `\a`.  
I have made som-rs to adhere to that behavior but it intrigued me when I saw it.  
I wonder if it is a strict requirement or an inefficacy that is left to implementations to decide how to handle.

**Comment by smarr** [comment]
> An interesting thing about string escapes in SOM that I found surprising is that, in case of a unknown escaped character (like `\a`), implementations seems to silently discard the character

That doesn't sound like it's intended and sounds rather like a bug.
I'd probably trigger a parse error for these cases.

**Comment by ltratt** [comment]
yksom throws an error in such a situation (as @smarr suggests).

**Comment by vext01** [comment]
One interpretation of `\a` would be the terminal bell :P

**Comment by ltratt** [comment]
AFAIK the only documentation for string escapes is here https://github.com/SOM-st/SOM/blob/master/TestSuite/StringTest.som#L94. No system bell I'm afraid ;)

File: src/lib/compiler/ast_to_instrs.rs [hunk 1]
```diff
@@ -658,9 +658,36 @@ impl<'a, 'input> Compiler<'a, 'input> {
             ast::Expr::String(span) => {
                 // XXX are there string escaping rules we need to take account of?
                 let s_orig = self.lexer.span_str(*span);
-                // Strip off the beginning/end quotes.
```
Comment (vext01) [inline src/lib/compiler/ast_to_instrs.rs:661]: Do we need to remove the XXX two lines above too?

**Comment by vext01** [comment]
Just one comment.

File: src/lib/compiler/ast_to_instrs.rs (hunk 1 above)
Comment (ltratt) [reply src/lib/compiler/ast_to_instrs.rs:661]: Oops, fixed in https://github.com/softdevteam/yksom/pull/154/commits/dc6f2ec5a437813e9498eedd091e9acb23483e9f.

**Comment by vext01** [comment]
Please squash.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**